src/platform_layers/linux_platform_layer/src/linux_adu_core_exports.cpp

```diff
@@ -68,14 +69,26 @@ void ADUC_Unregister(ADUC_Token token)
  */
 int ADUC_RebootSystem()
 {
+    int exitStatus = 0;
+    std::string output;
     Log_Info("ADUC_RebootSystem called. Rebooting system.");
 
     // Commit buffer cache to disk.
     sync();
 
-    std::string output;
-    std::vector<std::string> args{ "--update-type", "common", "--update-action", "reboot" };
-    const int exitStatus = ADUC_LaunchChildProcess(ADUSHELL_FILE_PATH, args, output);
+    const ADUC_ConfigInfo* config = ADUC_ConfigInfo_GetInstance();
+    if (config == NULL)
+    {
+        Log_Error("Failed to get config info instance.");
+        exitStatus = 126; // Command invoked cannot execute.
```

Comment:
magic number. where is 126 coming from? it is supposed to be ENOKEY (126)?  https://github.com/torvalds/linux/blob/master/include/uapi/asm-generic/errno.h#LL110C9-L110C9
Should it be ENOMEM instead?  #include <errno.h> and use symbol

Reply:
Fix